src/Components/Server/src/Circuits/DefaultCircuitFactory.cs

```diff
@@ -44,13 +45,15 @@ public override CircuitHost CreateCircuitHost(HttpContext httpContext, IClientPr
             }
 
             var scope = _scopeFactory.CreateScope();
-            var jsRuntime = new RemoteJSRuntime(client);
+            var environment = scope.ServiceProvider.GetRequiredService<ComponentEnvironment>();
+            InitializeEnvironment(environment, client);
```

Comment:
It's really unusual to see state on a service being mutated after initialization. If the intent is to tell components where they're being rendered, should it be on the `Renderer`?

Reply:
No, we could do it all in services, but some of these things require access to the httpcontext